src/components/tiles/table/use-content-change-handlers.ts

```diff
@@ -124,22 +125,36 @@ export const useContentChangeHandlers = ({
     if (!readOnly && consumerTile) {
       const sharedModelManager = consumerTile.tileEnv?.sharedModelManager;
       if (sharedModelManager?.isReady) {
+        // If the consumer tile does not support multiple shared data sets, remove it from
+        // any existing shared data set before linking.
+        if (!getTileContentInfo(consumerTile.type)?.supportsMultipleDataSets) {
+          const existingSharedTable = sharedModelManager?.findFirstSharedModelByType(SharedDataSet, tileInfo.id);
+          if (existingSharedTable) {
+            sharedModelManager?.removeTileSharedModel(consumerTile, existingSharedTable);
+          }
+        }
         const sharedTable = sharedModelManager?.findFirstSharedModelByType(SharedDataSet, modelRef.current.id);
         sharedTable && sharedModelManager?.addTileSharedModel(consumerTile, sharedTable);
       }
     }
   }, [getContent, readOnly, modelRef]);
 
   const unlinkTile = useCallback((tileInfo: ITileLinkMetadata) => {
-    const consumerTile = getTileContentById(getContent(), tileInfo.id);
-    if (!readOnly && consumerTile) {
-      const sharedModelManager = consumerTile.tileEnv?.sharedModelManager;
+    const linkedTile = getTileContentById(getContent(), tileInfo.id);
```

Comment:
I changed `consumerTile` to `linkedTile` here because in some cases, like when a table is linked to another table, it's possible that the other tile is the provider, not the consumer.